device/transport/amqp/src/amqp.ts

```diff
@@ -387,10 +395,12 @@ export class Amqp extends EventEmitter implements Client.Transport {
             /*Codes_SRS_NODE_DEVICE_AMQP_16_042: [The `disableMethods` method shall call `detach` on the device method links and call its callback when these are successfully detached.]*/
             /*Codes_SRS_NODE_DEVICE_AMQP_16_043: [The `disableMethods` method shall call its `callback` with an `Error` if it fails to detach the device method links.]*/
             this._deviceMethodClient.detach(callback);
-          }
+          },
+          /*Codes_SRS_NODE_DEVICE_AMQP_16_082: [if the handler specified in the `setDisconnectHandler` call is called while the `Amqp` object is connected, the connection shall be disconnected and an `disconnect` event shall be emitted with the error translated to a transport-agnostic error.]*/
+          disconnectError: (err, callback) => this._fsm.transition('disconnecting', err, callback)
```

Comment:
>disconnectError [](start = 10, length = 15)

same question on what state we should transition into.

Reply:
same answer then :)

---
In reply to: [175764784](https://github.com/Azure/azure-iot-sdk-node/pull/254#discussion_r175764784) [](ancestors = 175764784)